src/components/DetailsPanel/index.js

```diff
@@ -7,52 +7,41 @@ import {
   Header, Right, Title, Left,
 } from 'native-base';
 import { SimpleLineIcons } from '@expo/vector-icons'; // eslint-disable-line import/no-extraneous-dependencies
-import { shape } from 'prop-types';
-import produce from 'immer';
+import { arrayOf, shape } from 'prop-types';
+import { connect } from 'react-redux';
 
 import Colors from '../../constants/Colors';
 import SortingHeader from './SortingHeader';
-import { SORT_ASC, SORT_DESC, sortFields } from '../../constants/sorting';
-import TypeGroupContainer from '../TypeGroupContainer';
+import { SORT_DESC, sortFields } from '../../constants/sorting';
+import DateAccordionsContainer from '../DateAccordionContainer/DateAccordionsContainer';
+import SubTypeAccordionsContainer from '../SubTypeAccordion/SubTypeAccordionsContainer';
+import { savedRecordsGroupedByTypeSelector } from '../../redux/selectors';
 
-const { RECORD_TYPE, RECORD_DATE, TIME_SAVED } = sortFields;
+const DetailsPanel = ({ navigation, collection, savedRecordsGroupedByType }) => {
+  const { savedRecordsSortingState: sortingState } = collection;
+  const { RECORD_TYPE, RECORD_DATE, TIME_SAVED } = sortFields;
 
-const defaultSortingState = {
-  activeSortField: 'record-type',
-  sortDirections: {
-    [RECORD_TYPE]: SORT_DESC,
-    [RECORD_DATE]: SORT_DESC,
-    [TIME_SAVED]: SORT_DESC,
-  },
-};
-
-const DetailsPanel = ({ navigation, collection }) => {
-  const [sortingState, setSortingState] = useState(defaultSortingState);
   const handlePressNoteIcon = () => {
     navigation.navigate('CollectionNotes');
   };
-  const handleSortChange = (sortField) => {
-    setSortingState((state) => produce(state, (draft) => {
-      if (state.activeSortField === sortField) {
-        const prevDir = state.sortDirections[sortField];
-        // eslint-disable-next-line no-param-reassign
-        draft.sortDirections[sortField] = (prevDir === SORT_ASC) ? SORT_DESC : SORT_ASC;
-      }
-      draft.activeSortField = sortField; // eslint-disable-line no-param-reassign
-    }));
-  };
 
   const displayAccordion = () => {
     switch (sortingState.activeSortField) {
       case RECORD_TYPE:
         return (
-          <TypeGroupContainer
+          <SubTypeAccordionsContainer
+            data={savedRecordsGroupedByType}
             isDescending={sortingState.sortDirections[RECORD_TYPE] === SORT_DESC}
             fromDetailsPanel
           />
         );
       case RECORD_DATE:
-        return <Text>RecordDate</Text>;
+        return (
+          <DateAccordionsContainer
+            isDescending={sortingState.sortDirections[RECORD_DATE] === SORT_DESC}
```

Comment:
for this and other components:  

Does `isDescending` still need to be prop-drilled, given that `sortingState` is now in Redux state?

Reply:
I figured importing `sortingState` once from `DetailsPanel` is better than the components getting them individually. Also i think it's better for "generic" components like `SubTypeAccordionContainer` to explicitly grab that slice of redux when it only needs it occasionally.